tests/sdk.spec.ts

```diff
@@ -70,64 +70,128 @@ describe('HuddlySDK', () => {
   });
 
   describe('#setupDeviceDiscoveryListeners', () => {
+    const dummySerial = '123456';
     const dummyGO = {
-      productId: 0x11
+      productId: 0x11,
+      serialNumber: dummySerial
     };
     const dummyIQ = {
-      productId: 0x21
+      productId: 0x21,
+      serialNumber: dummySerial
     };
     let huddlygoInitStub;
     let boxfishInitStub;
     let discoveryEmitter;
     let otherEmitter;
     let hidInterfaceStub;
+    let dummyTargetSerial;
+
+    function initSdk() {
+      new HuddlySdk(nodeusbDeviceApi, [nodeusbDeviceApi], { emitter: otherEmitter, apiDiscoveryEmitter: discoveryEmitter, serial: dummyTargetSerial });
+    }
+
     beforeEach(() => {
       huddlygoInitStub = sinon.stub(HuddlyGo.prototype, 'initialize').resolves();
       boxfishInitStub = sinon.stub(Boxfish.prototype, 'initialize').resolves();
       hidInterfaceStub = sinon.stub(DeviceFactory, 'getHIDInterface').resolves();
       discoveryEmitter = new EventEmitter();
       otherEmitter = new EventEmitter();
-      new HuddlySdk(nodeusbDeviceApi, [nodeusbDeviceApi], { emitter: otherEmitter, apiDiscoveryEmitter: discoveryEmitter });
+      dummyTargetSerial = undefined;
     });
     afterEach(() => {
       huddlygoInitStub.restore();
       boxfishInitStub.restore();
       hidInterfaceStub.restore();
+
     });
 
     it('should emit ATTACH event with Boxfish instance when attached device is boxfish', (done) => {
+      initSdk();
```

Comment:
you can call `initSdk()` on the `beforeEach` hook, in that way, `initSdk` will be called before each test